optimum/intel/ipex/modeling_base.py

```
@@ -240,129 +145,30 @@ def _from_pretrained(
-                The file name of the model to load. Overwrites the default file name and allows one to load the model
-                with a different name.
         """
-        if use_auth_token is not None:
-            warnings.warn(
-                "The `use_auth_token` argument is deprecated and will be removed in v5 of Transformers. Please use `token` instead.",
-                FutureWarning,
-            )
-            if token is not None:
-                raise ValueError(
-                    "Both the arguments `use_auth_token` and `token` were specified, which is not supported. Please specify only `token`."
-                )
-            token = use_auth_token
-
-        commit_hash = kwargs.pop("_commit_hash", None)
-
-        model_kwargs = {
-            "revision": revision,
-            "token": token,
-            "cache_dir": cache_dir,
-            "subfolder": subfolder,
-            "local_files_only": local_files_only,
-            "force_download": force_download,
-        }
-
-        if not getattr(config, "torchscript", False):
-            logger.warning("Detect torchscript is false. Convert to torchscript model!")
-
-            if is_torch_version("<", "2.1.0"):
-                raise ImportError("`torch>=2.0.0` is needed to trace your model")
-
-            task = cls.export_feature
-            config.torch_dtype = torch_dtype
-            model = TasksManager.get_model_from_task(
-                task,
-                model_id,
-                library_name="transformers",
-                trust_remote_code=trust_remote_code,
-                torch_dtype=torch_dtype,
-                _commit_hash=commit_hash,
-                **model_kwargs,
+        if getattr(config, "torchscript", False):
+            logger.warning(
+                "IPEXModel will not support torch script model in the future, fallback to TSModelForCausalLM"
             )
+            return TSModelForCausalLM.from_pretrained(model_id, **kwargs)
```

An instance of `TSModelForCausalLM` will be created for all `IPEXModel` (even for encoder models) which doesn't really make sense to me. Also it's not tested anywhere from what I see, I prefer to raise an error here instead of keeping support that we're not sure works / is compatible with the previous integration

---

done.